src/cpp/RiderLink/Source/RiderLink/Private/RiderLink.cpp

```diff
@@ -13,17 +14,27 @@ DEFINE_LOG_CATEGORY(FLogRiderLinkModule);
 
 IMPLEMENT_MODULE(FRiderLinkModule, RiderLink);
 
+static FString GetProjectName()
+{
+	FString ProjectNameNoExtension = FApp::GetProjectName();
+	if (ProjectNameNoExtension.IsEmpty())
+		ProjectNameNoExtension = TEXT("<ENGINE>");
+	return ProjectNameNoExtension;
+}
+
 void FRiderLinkModule::ShutdownModule()
 {
 	UE_LOG(FLogRiderLinkModule, Verbose, TEXT("RiderLink SHUTDOWN START"));
 	ModuleLifetimeDef.terminate();
+	ProtocolFactory.Reset();
 	UE_LOG(FLogRiderLinkModule, Verbose, TEXT("RiderLink SHUTDOWN FINISH"));
 }
 
 void FRiderLinkModule::StartupModule()
 {
 	UE_LOG(FLogRiderLinkModule, Verbose, TEXT("RiderLink STARTUP START"));
-	ProtocolFactory::InitRdLogging();
+	ProtocolFactory = MakeUnique<class ProtocolFactory>(GetProjectName());
+	ProtocolFactory->InitRdLogging();
```

Comment:
If it's a class now and this method is not used anywhere, mb it's better to be moved to the constructor?